### PR TITLE
feat: enable GORACE=halt_on_panic=1 in machined binary

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: https://github.com/talos-systems/talos/issues/3533

### CODE REVIEW

File: internal/app/machined/main.go [hunk 1]
```diff
@@ -192,6 +192,14 @@ func run() error {
 		return errors.New("error setting PATH")
 	}
 
+	if runtime.WithRace {
```
[the remark this answers is not in the document]
Comment (Unix4ever) [reply internal/app/machined/main.go:195]: yeah, if it doesn't work in the binary, this approach won't work...

[the remark this answers is not in the document]
Comment (Unix4ever) [reply internal/app/machined/main.go]: updated

**Review by smira — APPROVED** [review]

**Pull request merged**